### DISCUSSION TITLE
Configuration for windows global system proxy

### DISCUSSION THREAD

**Discussion by feelingnothing** [shadowsocks/shadowsocks-rust#1650] [Q&A] [answered]
First thing I tried is setting up firefox to use socks5. Despite the proxy working (tested with simple script using `reqwest`) firefox did not seem to care. After that I tried to add local socks proxy as global in window's settings. That didn't work either. Then I tried shadowsocks client written in c#, that worked for first request I sent in firefox, then it just didn't seem to work ever again.
Next thing I tried is `wintun`, after downloading the dll, building the project from source with `local-tun` feature enabled and launching `sslocal` with these parameters: `sslocal --protocol tun --outbound-bind-interface "Ethernet"` - nothing happened.
I see that it created the wintun network adapter, but my ip address did not change.
Logs: https://gist.github.com/feelingnothing/6736393b10d5bb47ee23061fe344b339
I'm already 2 days deep into this, maybe I'm just in the wrong project and sslocal is not designed to be global system-wide proxy.
My windows version: 23H2(OS Build 22631.4112)
Wintun Version: 0.14.1

**Comment by feelingnothing** [comment]
Also tried https://github.com/spyophobia/shadowsocks-gtk-rs, after fixing it not compiling it did not work too.

**Comment by zonyitoo** [comment, marked as answer]
SOCKS Proxy in Windows' network setting will be applied globally, maybe Firefox doesn't use that for making requests. It would be easier for you to set proxy setting in each different application with the specific SOCKS Proxy address.

For `local-tun`, it would create a new network adapter and listening on that. You should set the route table of Windows to route all the network packets to this new network adapter to make it work.

Tools in this project is the fundamental service of shadowsocks. What you are looking for is a one-click application that help you setup all the necessary parts, which is not the design goal of this project.

**Reply by feelingnothing** [reply]
Thanks for clarification.
